samples/csharp/getting-started/Ranking_AutoML/Ranking/Ranking/Program.cs

```diff
@@ -0,0 +1,234 @@
+            Console.WriteLine("\n===== Refitting on train+valid+test to get the final model to launch to production =====");
+            var trainPlusValidationPlusTestDataView = textLoader.Load(new MultiFileSource(TrainDatasetPath, ValidationDatasetPath, TestDatasetPath));
+
+            var refitModelWithValidationSet = experimentResult.BestRun.Estimator.Fit(trainPlusValidationPlusTestDataView);
+
+            IDataView predictionsRefitOnTrainValidationPlusTest = refitModelWithValidationSet.Transform(testDataView);
+
+            var metricsRefitOnTrainValidationPlusTest = mlContext.Ranking.Evaluate(predictionsRefitOnTrainValidationPlusTest, rankingEvaluatorOptions);
+
+            ConsoleHelper.PrintRankingMetrics(experimentResult.BestRun.TrainerName, metricsRefitOnTrainValidationPlusTest);
+
+            // STEP 5: Evaluate the model and print metrics
+            ConsoleHelper.ConsoleWriteHeader("=============== Evaluating model's nDCG with test data ===============");
+            RunDetail<RankingMetrics> bestRun = experimentResult.BestRun;
+
+            ITransformer trainedModel = bestRun.Model;
+            _predictions = trainedModel.Transform(testDataView);
+
+            var metrics = mlContext.Ranking.Evaluate(_predictions, rankingEvaluatorOptions);
+
+            ConsoleHelper.PrintRankingMetrics(bestRun.TrainerName, metrics);
+
+            // STEP 6: Save/persist the trained model to a .ZIP file
+            mlContext.Model.Save(trainedModel, trainDataView.Schema, ModelPath);
+
+            Console.WriteLine("The model is saved to {0}", ModelPath);
+
+            return trainedModel;
+        }
+
+        private static void TestSinglePrediction(MLContext mlContext)
+        {
+            ConsoleHelper.ConsoleWriteHeader("=============== Testing prediction engine ===============");
+
+            ITransformer trainedModel = mlContext.Model.Load(ModelPath, out var modelInputSchema);
+            Console.WriteLine($"=============== Loaded Model OK  ===============");
+
+            // In the predictions, get the scores of the search results included in the first query (e.g. group).
+            var searchQueries = mlContext.Data.CreateEnumerable<RankingPrediction>(_predictions, reuseRowObject: false);
+            var firstGroupId = searchQueries.First().GroupId;
+            var firstGroupPredictions = searchQueries.Take(100).Where(p => p.GroupId == firstGroupId).OrderByDescending(p => p.Score).ToList();
+
+            // The individual scores themselves are NOT a useful measure of result quality; instead, they are only useful as a relative measure to other scores in the group. 
+            // The scores are used to determine the ranking where a higher score indicates a higher ranking versus another candidate result.
+            foreach (var prediction in firstGroupPredictions)
+            {
+                Console.WriteLine($"GroupId: {prediction.GroupId}, Score: {prediction.Score}");
```

Comment:
This prints:
```md
=============== Testing prediction engine ===============
#########################################################
=============== Loaded Model OK  ===============
GroupId: 94629043, Score: 10.5748415
GroupId: 94629043, Score: 8.747048
GroupId: 94629043, Score: 8.339484
GroupId: 94629043, Score: 8.295864
GroupId: 94629043, Score: 7.7126718
GroupId: 94629043, Score: 7.094361
GroupId: 94629043, Score: 6.403935
GroupId: 94629043, Score: 5.6126056
GroupId: 94629043, Score: 5.4343157
GroupId: 94629043, Score: 5.341767
GroupId: 94629043, Score: 3.4955919
GroupId: 94629043, Score: 3.0068336
GroupId: 94629043, Score: 2.925596
...
```

Is there more information we can print? Currently the user has no way to compare the ranked results against the original input data.

Could we print the correct/input Label in a column? The user could then verify the input Label value is decreasing in the list.

Reply:
I can add the original label to the output. Would that help?
![image](https://user-images.githubusercontent.com/1578160/102887985-e5835400-4425-11eb-9bfe-52bd5c84b0bf.png)

Reply:
Is there a way to do a late merge, without adding `Label` to the `RankingPrediction` class? 

Do you think if we add the `Label` to the output class, some users will misconstrue it as the predicted label (`Score` is the actual prediction)? Seeing no mispredictions, users could assume their model is perfect, and launch it to production only to find the `Label` field is then empty.

Even without the added label output, this PR is very good. I'm ok merging as is, or with adding the correct/input label.

Reply:
My only thought is to do something like this:

```csharp
var predictionsPreview = _predictions.Preview();

for (int i = 0; i < firstGroupPredictions.Count; i++)
{
  var currentPredictionPreview = predictionsPreview.RowView[i].Values;

  Console.WriteLine($"GroupId: {firstGroupPredictions[i].GroupId}, Score: {firstGroupPredictions[i].Score}");
}
```

Use the predictions from the test set and get the label from the `Preview` of it.

I'm not sure if it's the best thing to do or not, though. If this isn't good, perhaps we can get this in and can do another update once we can find a better solution.

Reply:
How does this look:
```c#
// Label values from the test dataset (not the predicted scores/labels)
IEnumerator<string> labelEnumerator = mlContext.Data.CreateEnumerable<RankingData>(testDataView, true).Select(a => a.Label).GetEnumerator();

foreach (var prediction in firstGroupPredictions)
{
    labelEnumerator.MoveNext();
    Console.WriteLine($"GroupId: {prediction.GroupId}, Score: {prediction.Score}, Correct Label: {labelEnumerator.Current}");
}
```

It would require passing in `testDataView`. Technically, the same enumerator can be made from `firstGroupPredictions`, though reading from the test dataset reenforces that the printed label is not part of the prediction, but instead a given label.

Reply:
That works great! Pushed the latest changes with this in it. Thanks!

Reply:
LGTM. Good to merge?

Reply:
I think so! Thank you for helping me with this sample!